src/states/FunctionState.ts

```diff
@@ -130,11 +130,19 @@ export class FunctionState implements ContextState {
             }
         }
 
-        if (property === SubstituteMethods.returns) {
+        if (property === SubstituteMethods.returns
+            || property === SubstituteMethods.resolves
+            || property === SubstituteMethods.rejects
+        ) {
             return (...returns: any[]) => {
                 if (!this._lastArgs) {
                     throw SubstituteException.generic('Eh, there\'s a bug, no args recorded for this return :/')
                 }
+                if (property === SubstituteMethods.resolves) {
+                    returns = returns.map(value => Promise.resolve(value))
+                } else if(property === SubstituteMethods.rejects) {
+                    returns = returns.map(value => Promise.reject(value))
+                }
                 this.returns.push({
                     returnValues: returns,
                     returnIndex: 0,

```

Comment:
Nice how you refactored it! But I don't think it's a good idea to reassigning variables, I think something like this would be better:
```suggestion
const returnMock: Partial<ReturnMock> = { returnIndex: 0, args: this._lastArgs };
                switch (property) {
                    case SubstituteMethods.returns:
                        returnMock.returnValues = returns;
                        break;
                    case SubstituteMethods.resolves:
                        returnMock.returnValues = returns.map(value => Promise.resolve(value));
                        break;
                    case SubstituteMethods.rejects:
                        returnMock.returnValues = returns.map(value => Promise.reject(value));
                        break;
                    default:
                        throw SubstituteException.generic(
                            `Expected one of the following methods: "${SubstituteMethods.returns}", "${SubstituteMethods.resolves}" or "${SubstituteMethods.rejects}"`
                        );
                }

                this.returns.push(<ReturnMock>returnMock);
```

Reply:
Nice! I just removed the `default:` `throw` from the code. It will never be reached in this case because of the `if` condition above :)

Reply:
True, the if prevents the default case. I added it in the suggestion for the future, in case new methods are added and the switch case doesn't get updated. It makes it easier to debug